datajunction-server/datajunction_server/utils.py

```diff
@@ -261,4 +261,32 @@ async def get_current_user(request: Request) -> Optional["User"]:
     return None  # pragma: no cover
 
 
+async def get_current_user_and_upsert(
```

Comment:
This looks good. But why not use more english here? Say `get_and update_current_user` ?

Reply:
Updated!